gpt_index/embeddings/base.py

```diff
@@ -33,11 +33,27 @@ def __init__(self) -> None:
     def _get_query_embedding(self, query: str) -> List[float]:
         """Get query embedding."""
 
-    def get_query_embedding(self, query: str) -> List[float]:
+    def _combine_embeddings(self, embeddings: List[List[float]]) -> List[float]:
+        return list(np.array(embeddings).mean(axis=0))
```

Comment:
out of curiosity, are there other means of combining embeddings? (maybe i need to take a look at your future PR's)

Reply:
Currently, we just have the naive approach of taking the mean embedding (which is what HyDE paper does).

IMO this is one downside of implementing HyDE as a query processor (str -> str) rather than a embedder (str -> embedding). Namely that the configuration of how to embed one (or multiple) hypothetical documents would be separate from the document generation logic.